tools/acorn-optimizer.js

```diff
@@ -23,6 +23,19 @@ function assert(condition, text) {
   if (!condition) throw text + ' : ' + new Error().stack;
 }
 
+function warnOnce(a, msg) {
```

Comment:
I don't understand this pattern.   Won't msg always be undefined here?

Reply:
In the current use, yes. But it's an optional param that might be useful later. (I just copied this utility function from utility.js.)

Reply:
I still don't understand what `a` means here?  What is the point of it?

Reply:
It's an optional check - if both params are provided, the first is a boolean. if it's false, we warn. Like an assert, but a warning.

I'll remove that part as it does look confusing now that you mention it, and doesn't help atm.